### PR TITLE
consolidate onboarding between mobile app and extension

### PR DESCRIPTION
**Pull request by peterpme**
creates an OnboardingProvider that can re-use the majority of the different functions and state while retaining the separate flows

this is an attempt at keeping the mobile app in sync with the extension by consolidating the logic

### CODE REVIEW

File: packages/app-extension/src/components/Onboarding/pages/KeyringTypeSelector.tsx [hunk 1]
```diff
@@ -9,7 +9,7 @@ export const KeyringTypeSelector = ({
   action,
   onNext,
 }: {
-  action: "create" | "import" | "recover";
+  action: "create" | "import" | "recover" | string;
```
Comment (tomlinton) [inline packages/app-extension/src/components/Onboarding/pages/KeyringTypeSelector.tsx:12]: What is string for?

Comment (peterpme) [reply packages/app-extension/src/components/Onboarding/pages/KeyringTypeSelector.tsx:12]: typechecking -_-

**Comment by hkirat** [comment]
This is super cool, much needed to minimise code repetition b/w app and web.
I was looking at https://github.com/jitsi/jitsi-meet 's code recently and they also seem to have similar abstractions to minimize code repetition across mobile and web

**Comment by peterpme** [comment]
Thanks @hkirat :)

File: packages/app-extension/src/components/Onboarding/pages/KeyringTypeSelector.tsx (hunk 1 above)
Comment (tomlinton) [reply packages/app-extension/src/components/Onboarding/pages/KeyringTypeSelector.tsx:12]: Elaborate please sir 😛

**Pull request merged**